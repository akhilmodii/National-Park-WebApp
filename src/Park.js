import React, { Component } from 'react';
import Buttons from './Buttons.js';
import { connect } from 'react-redux';
import { toggleVisitedPark, toggleBucketListPark, setParkCardToShow } from './actions/index.js';
import {Dialog} from "./Dialog";

class Park extends Component {
 constructor(props) {
    super(props);
    this.state = {
      displayFull: false  
   };
 }
  toggleFullCard= () => {
    this.setState({
      displayFull: !this.state.displayFull
    });
  }

  removeCard = (event) => {
    event.preventDefault();
    this.props.setParkCardToShow('');
    this.setState({
      displayFull: false,
    });
  }

  setStateText = () => {
    const { states } = this.props.park;
    if (states.length === 1) {
      return `State: ${states[0]}`
    } else {
      return states.reduce((acc, state, index) => {
        acc = index === 0 ? `States: ${state}` : acc + `, ${state}`;
        return acc;
      }, '');
    }
  }

  render() {
    const { name, images, url, description, weatherInfo, directionsInfo, entranceFees, activities} = this.props.park;
    const { displayFull } = this.state;
    const stateText = this.setStateText();
    // const {eFees} = entranceFees.map((d) => <li>key={d.cost}>{d.cost}</li>
    return (
      <div className="card-overlay">


          <div className={displayFull ? "park-card-large" : "park-card-small"}>
          <div className={displayFull ? "park-text-large" : ""}>
            <i className="far fa-times-circle" onClick={this.removeCard}></i>

            <h1 className="park-title">{name} National Park</h1>
            {
              !displayFull && <img className="park-img-small" alt="park" src={images[0].url} />
            }

            <h3>{stateText}</h3>
            <a className="nps-link" href={url} target="_blank">Link to {name}'s National Park Service Page</a>
              {/*<h2>Park Description:</h2>*/}
              <h3 className="park-descrip"> <h2>Park Description:</h2> {description}</h3>

              <h3 className="weather-info"> <h2>Weather Information:</h2>{weatherInfo}</h3>
              <h3 className="directions"><h2>Directions to the Park:</h2> {directionsInfo}</h3>
              {/*<h3>*/}
              {/*    {entranceFees.map(function(d){*/}
              {/*        return (<li>{d.cost}</li>)*/}
              {/*    })}*/}
              {/*</h3>*/}

              <h3 className= {"activities"}> <h2>Park Activities:</h2>
                  {activities.map(function(d){
                      return (<li>{d.name}</li>)
                  })}
              </h3>
            <button className="button-small" onClick={this.toggleFullCard}>{displayFull ? "View Less" : "View More"}</button>
              <div className="user-list-btns">
                  <Buttons
                      iconType="fas fa-hiking"
                      storageKey="visited"
                  />
                  <Buttons
                      iconType="fas fa-clipboard-list"
                      storageKey="bucket"
                  />
                  <div>
                      <button onClick={(e) => this.setState({isOpen: true})} id="show-bucket-button"> Bookings </button>
                      <Dialog isOpen={this.state.isOpen} onClose={(e) => this.setState({isOpen: false})}>
                          <h2>Bookings</h2>
                          <li><a href={"https://www.tripadvisor.com/Hotels-g60999-Yellowstone_National_Park_Wyoming-Hotels.html\n"}>Hotels</a></li>
                          <li><a href={"https://www.tripadvisor.com/Restaurants-g60999-Yellowstone_National_Park_Wyoming.html\n"}>Restaurants</a></li>
                          <li><a href={"https://www.tripadvisor.com/Attractions-g60999-Activities-Yellowstone_National_Park_Wyoming.html\n"}>Things to Do</a></li>
                          <li><a href={"https://www.tripadvisor.com/Flights-g60999-Yellowstone_National_Park_Wyoming-Cheap_Discount_Airfares.html\n"}>Travel Forums</a></li>
                          <li><a href={"https://www.tripadvisor.com/ShowForum-g60999-i481-Yellowstone_National_Park_Wyoming.html\n"}>Flights</a></li>
                      </Dialog>
                  </div>
                  {/*<Buttons*/}
                  {/*    iconType="fas fa-booking"*/}
                  {/*    storageKey="visited"*/}
                  {/*/>*/}
              </div>
          </div>
          {
            displayFull && <img className="park-img-large" alt="park" src={images[images.length-1].url} />
          }

        </div>

      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  toggleVisitedPark: (parkCode) => dispatch(toggleVisitedPark(parkCode)),
  toggleBucketListPark: (parkCode) => dispatch(toggleBucketListPark(parkCode)),
  setParkCardToShow: (parkCode) => dispatch(setParkCardToShow(parkCode)),
});

export default connect(null, mapDispatchToProps)(Park);