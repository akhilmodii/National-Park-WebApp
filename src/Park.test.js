import React from 'react';
import Park from './Park';
import { shallow } from 'enzyme';

const selectedPark = {
  parkName: "Death Valley",
  state: ["California", "Nevada"],
  dateEstablished: "October 31, 1994",
  latitude: "36.24°N",
  longitude: "116.82°W",
  annualVisitors: 1294827,
  petsWelcome: true,
  editorsChoice: "Badwater Basin Salt Flats",
  websiteUrl: "https://www.nps.gov/deva/index.htm",
  urlCode: "deva",
  image: "assets/death-valley.jpg",
  imageRef: "<a rel=\"nofollow\" class=\"external text\" href=\"https://sites.google.com/site/thebrockeninglory/\">Brocken Inaglory</a> - <span class=\"int-own-work\" lang=\"en\">Own work</span>, <a href=\"https://creativecommons.org/licenses/by-sa/3.0\" title=\"Creative Commons Attribution-Share Alike 3.0\">CC BY-SA 3.0</a>, <a href=\"https://commons.wikimedia.org/w/index.php?curid=5990426\">Link</a>",
  sizeAcres: 337306314,
  description: "The Badlands are a collection of buttes, pinnacles, spires, and mixed-grass prairies. The White River Badlands contain the largest assemblage of known late Eocene and Oligocene mammal fossils. The wildlife includes bison, bighorn sheep, black-footed ferrets, and prairie dogs."
};
const visitedParks = ['deva', 'grsm'];
const bucketListParks = ['yell'];
const removeCardMock = jest.fn();
const updateParkCodesMock = jest.fn();

describe('Park', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Park
        selectedPark={selectedPark}
        removeCard={removeCardMock}
        visitedParks={visitedParks}
        bucketListParks={bucketListParks}
        updateParkCodes={updateParkCodesMock}
      />
    );
  });
  
  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
    expect(wrapper.state()).toEqual({ displayFull: false });
  });

  it('should be able to toggle between small and large card', () => {
    expect(wrapper.state('displayFull')).toEqual(false);
    wrapper.find('.button-small').simulate('click');
    expect(wrapper.state('displayFull')).toEqual(true);
  });

  it('should be able to remove a park card', () => {
    wrapper.find('#remove-card').simulate('click', { preventDefault: () => { } });
    expect(removeCardMock).toBeCalled();
    expect(wrapper.state('displayFull')).toEqual(false);
  });
});