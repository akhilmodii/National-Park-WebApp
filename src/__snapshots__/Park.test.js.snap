// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Park should match the snapshot with all data passed in 1`] = `
<div
  className="card-overlay"
>
  <div
    className="park-card-small"
  >
    <i
      className="far fa-times-circle"
      id="remove-card"
      onClick={[Function]}
    />
    <h1
      className="park-title"
    >
      Death Valley
       National Park
    </h1>
    <img
      alt="park"
      className="park-img-small"
      src="./assets/death-valley.jpg"
    />
    <h3
      className="park-text-small"
    >
      Park Highlight: 
      Badwater Basin Salt Flats
    </h3>
    <button
      className="button-small"
      onClick={[Function]}
    >
      View More
    </button>
    <div
      className="user-list-btns"
    >
      <Buttons
        bucketListParks={
          Array [
            "yell",
          ]
        }
        iconType="fas fa-hiking"
        parkUrl="deva"
        storageKey="visitedParks"
        updateParkCodes={[Function]}
        visitedParks={
          Array [
            "deva",
            "grsm",
          ]
        }
      />
      <Buttons
        bucketListParks={
          Array [
            "yell",
          ]
        }
        iconType="fas fa-clipboard-list"
        parkUrl="deva"
        storageKey="bucketList"
        updateParkCodes={[Function]}
        visitedParks={
          Array [
            "deva",
            "grsm",
          ]
        }
      />
    </div>
  </div>
</div>
`;
